src/components/structures/MatrixChat.tsx

```diff
@@ -1258,8 +1260,7 @@ export default class MatrixChat extends React.PureComponent<IProps, IState> {
         this.themeWatcher.recheck();
         StorageManager.tryPersistStorage();
 
-        const cli = MatrixClientPeg.get();
-        createLocalNotificationSettingsIfNeeded(cli);
+        this.cli.on(ClientEvent.Sync, this.onInitialSync);
```

Comment:
This method doesn't do what it says it does, `ClientEvent.Sync` fires for the cached sync accumulator too, so this will not await an initial sync without the `if (prevState === "PREPARED" && state === "SYNCING") {` check